### ISSUE TITLE
Typescript Support

### ISSUE THREAD

**Issue by Slynchy** [georgealways/lil-gui#15]
Hi! I'd love to use `lil-gui` in my projects, but it's currently missing type definitions for TypeScript to fully work. I've not personally created type definitions for JavaScript projects before, but it might be something to consider to be able support TS projects.

![image](https://user-images.githubusercontent.com/7141218/142046177-839b95d3-91e4-4729-983a-97d095509f71.png)

**Comment by georgealways** [comment]
Hey! Yes, it would be nice to provide those typings. I'm not a TypeScript user personally, so I don't really know how that's done, but would gladly welcome any help! Is there an automated way to generate that through the JSDoc comments maybe? 

(Side note: the JSDoc comments are mostly there to generate the API docs, but I was actually very pleasantly surprised by how much auto-complete VSCode was offering me in a plain JS project from the JSDoc comments alone.)

**Comment by Slynchy** [comment]
I did some searching and it seems you _can_ use JSDoc comments to generate `.d.ts` files, but apparently it's just as simple as running the Typescript compiler against your existing JS? Check it out: https://www.typescriptlang.org/docs/handbook/declaration-files/dts-from-js.html

**Comment by georgealways** [comment]
Very cool—Is the `d.ts` something that gets incorporated into this repo? Or is that a separate package that gets published?

I'm not sure why the build task is crashing for you, it should run `sass` before it runs `rollup`. What's the exact error you're getting? Also you can just run `build` for the library itself, `build:all` does the homepage and API.

**Comment by Slynchy** [comment]
Yes, the `.d.ts` files should be in the same folder as their accompanying `.js` files. You can also create another `npm` module under the name `@types/lil-gui` but I think this is a legacy way of doing it.

Full error below:
![image](https://user-images.githubusercontent.com/7141218/142629123-f487a732-4da7-4ac5-8fdf-12a67461b672.png)

**Comment by georgealways** [comment]
Cool—I read the guide you linked and it seems pretty straightforward. I guess in theory we could make that part of the pre-publish flow. There's two parts I'm not crazy about though:

1. Adding `typescript` as a devDependency (don't want people to think this is a TS lib)
2. I wouldn't really be able to verify if they were generated correctly

I want to look into how other libs are handling this, but it should probably be provided one way or another.

(I'm going to open up another issue for the build error, it's strange)

**Comment by Slynchy** [comment]
That's probably an argument in favor of having a separate `@types/*` module, but I don't know anything about how these are published. Here's some docs for us both to read 😂 : https://www.typescriptlang.org/docs/handbook/declaration-files/publishing.html

**Comment by georgealways** [comment]
I just bumped the version and included a `d.ts` file in the dist folder (pointing to it with a `types` field in package.json). I was able to see some type information and JSDoc comments when I made a test project in typescript. Let me know if it works for you.

**Comment by Slynchy** [comment]
Hi, sorry! Busy week, will try drop-in replacing `dat.gui` sometime this weekend and let you know how I get on 👍
